js/src/api/pubsub/pubsub.js

```diff
@@ -0,0 +1,605 @@
+// Copyright 2015-2017 Parity Technologies (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+import { inAddress, inBlockNumber, inData, inHex, inNumber16, inOptions, inFilter, inDeriveHash, inDeriveIndex } from '../format/input';
+import { outAccountInfo, outAddress, outBlock, outChainStatus, outHistogram, outHwAccountInfo, outNodeKind, outNumber, outPeers, outTransaction, outSyncing, outReceipt, outLog, outAddresses, outRecentDapps, outVaultMeta } from '../format/output';
+
+import PubsubBase from './pubsubBase';
+
+export default class Pubsub extends PubsubBase {
+  constructor (transport) {
+    super(transport);
+    this._api = 'parity_subscribe';
+  }
+
+  unsubscribe (subscriptionIds) {
+    return this.removeListener(subscriptionIds);
+  }
+
+  // eth API
+  // `newHeads`, `logs`, `newPendingTransactions`, `syncing`
+  newHeads (callback) {
+    return this.addListener('eth_subscribe', 'newHeads', callback);
+  }
+  //
+  // logs (callback) {
+  //   throw Error('not supported yet');
+  // }
+  //
+  // newPendingTransactions (callback) {
+  //   throw Error('not supported yet');
+  // }
+  //
+  // syncing (callback) {
+  //   throw Error('not supported yet');
+  // }
+
+  // parity API
+  accountsInfo (callback) {
```

Comment:
We could, I had them split before but that makes it more difficult for the user again to keep track which subscriptionId belongs to which api for unsubscribing. 
We could split them again but keep that the Id's are non-correlating.
Also `eth`  only supports one method so far so the extra work for the user seemed unnecessary.